### ISSUE TITLE
[explicit-module-boundary-types] Requires return type on overloaded constructors

### ISSUE THREAD

**Issue by sveyret** [typescript-eslint/typescript-eslint#2163]
**Repro**

```JSON
{
  "extends": [
    "eslint:recommended",
    "plugin:@typescript-eslint/eslint-recommended",
    "plugin:@typescript-eslint/recommended",
    "prettier",
    "prettier/@typescript-eslint"
  ],
  "rules": {
    "@typescript-eslint/explicit-function-return-type": "off"
  }
}
```

```TS
export default class MyClass {
  public constructor()
  public constructor(
    specialKeyPrefix: string
  )
  public constructor(
    private readonly specialKeyPrefix: string = '.'
  ) {}
}
```

**Expected Result**

No warning.

**Actual Result**

Both constructor declarations (not the implementation) are emitting the following warning:

    2:21  warning  Missing return type on function  @typescript-eslint/explicit-module-boundary-types
    3:21  warning  Missing return type on function  @typescript-eslint/explicit-module-boundary-types

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.1.0` |
| `@typescript-eslint/parser`        | `3.1.0` |
| `TypeScript`                       | `3.9.3` |
| `ESLint`                           | `7.1.0` |
| `node`                             | `14.2.0` |

**Comment by bradzacher** [comment]
See #2150

Please use the search next time

**Comment by sveyret** [comment]
Sorry for the duplicate. I used search, but probably missed the issue because already closed.
